packages/gatsby-source-wordpress/src/gatsby-browser.ts

```diff
@@ -34,6 +33,23 @@ export function onRouteUpdate(): void {
   }
 }
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+let ReactDOM: any
+let hasWarnedReact17 = false
+
+try {
+  ReactDOM = require(`react-dom/client`)
+} catch (e) {
+  if (process.env.NODE_ENV === `development` && !hasWarnedReact17) {
```

Comment:
Warning against it seems incorrect? Users should still be able to use React 17 and React 18.

We handle it in other places like this: https://github.com/gatsbyjs/gatsby/blob/e3d57c0fb8ab8a89b29fd613a519340574b1239a/packages/gatsby/cache-dir/app.js#L43-L60

Reply:
Oh nice, I didn't know about `HAS_REACT_18`. Is that available in gatsby-browser?

Reply:
I added the message because the failed require outputs a warning to the terminal even though I'm catching the error

Reply:
Awesome, it is indeed available! Thanks @LekoArts !

Reply:
https://github.com/gatsbyjs/gatsby/pull/35707/commits/58a644030e09088bac041df2367fda9f78ad4e9d